bcda/bcdacli/cli.go

```diff
@@ -658,3 +680,74 @@ func setBlacklistState(cmsID string, blacklistState bool) error {
 
 	return db.Model(&aco).Update("blacklisted", blacklistState).Error
 }
+
+// CCLF file name pattern and regex
+const cclfPattern = `((?:T|P)\.BCD\..*\.ZC\d*)Y(\d{2}\.D\d{6}\.T\d{7})`
+
+var cclfregex = regexp.MustCompile(cclfPattern)
+
+func RenameCCLF(name string) string {
+	return cclfregex.ReplaceAllString(name, "${1}R${2}")
+}
+
+func CloneCCLFZips(path string) error {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+
+	// Iterate through all cclf zip files in provided directory
+	for _, f := range files {
+		// Make sure to not clone non CCLF files in case the wrong directory is given
+		if !cclfregex.MatchString(f.Name()) {
+			log.Infof("Skipping file `%s`, does not match expected name... ", f.Name())
+			continue
+		}
+		fn, err := CloneCCLFZip(path, f.Name())
+		if err != nil {
+			return err
+		}
+		log.Infof("Created runout file: %s", fn)
+	}
+	return nil
+}
+
+func CloneCCLFZip(path, name string) (string, error) {
+	// Open source zip file for cloning
+	zr, err := zip.OpenReader(filepath.Join(path, name))
+	if err != nil {
+		return "", err
+	}
+	defer zr.Close()
+
+	// Create destination runout zip file with proper nomenclature
+	filename := RenameCCLF(name)
+	newf, err := os.Create(filepath.Join(path, filename))
+	if err != nil {
+		return "", err
+	}
+	defer newf.Close()
+
+	zw := zip.NewWriter(newf)
+	defer zw.Close()
+
+	// For each CCLF file in src zip, rename and write to dst zip
+	for _, f := range zr.File {
+		r, err := f.Open()
+		if err != nil {
+			return filename, err
+		}
+		defer r.Close()
```

Comment:
Something to note in the future with using `defer...`, if we have unbounded number of files in the zip (we don't for CCLF), we may end up in a bad state since do not close the file until the very end of the function. So we may end up running out of file descriptors (or something else (OOM?))

If there that is a concern and you still want to leverage defer, you can always wrap the entire code block in an anonymous function.

```
r, err := func() (string, error) {
                f.Open()
		if err != nil {
			return filename, err
		}
		defer r.Close()
		w, err := zw.Create(RenameCCLF(f.Name))
		if err != nil {
			return filename, err
		}
		_, err = io.Copy(w, r)
		if err != nil {
			return filename, err
		}()
```

It **does not apply to CCLF** since we know it should have at most 10 files.